build/builder.go

```diff
@@ -104,15 +104,34 @@ type Options struct {
 	changedOrRemovedFiles []string
 }
 
-// HashOptions creates a hash of the options that affect an index.
-func (o *Options) HashOptions() string {
+// HashOptions contains only the options in Options that upon modification leads to IndexState of IndexStateMismatch during the next index building.
+type HashOptions struct {
+	sizeMax          int
+	disableCTags     bool
+	cTags            string
+	cTagsMustSucceed bool
+	largeFiles       []string
+}
```

Comment:
I decided to make these members unexported since HashOptions is just for printing a hash value or printing a struct for message logs.